packages/athena/src/client.ts

```diff
@@ -16,30 +12,44 @@ import type {
   OperationResult,
   ReactiveAdapter,
 } from './types.js';
-import { prepareQuery } from './utils.js';
+import { prepareOperation } from './utils.js';
 
 export interface ClientArgs {
   url: string;
   id: IdFetcher;
-  fetchMiddleware?: Array<Middleware>;
+  fetch?: typeof fetch;
+  buildRequest?: BuildRequest;
   adapter: ReactiveAdapter;
 }
 
+export type BuildRequest = (request: GraphQLRequest) => RequestInit;
```

Comment:
Do we want `Request` (as in `new Request(...)`) or `RequestInit` (the second arg to `new Request(url, initHere)`)? Conceptually, I think there are more things that are enabled when providing a whole `Request` object than just returning the `RequestInit`...

Reply:
Additionally, if we just build the request, we can make the URL field optional. 


---

Actually, it's basically more like this:

```ts
interface ClientArgsCommon {
  id: IdFetcher;
  adapter: ReactiveAdapter;
}

interface ClientArgsWithBuildRequest extends ClientArgsCommon {
  buildRequest(GraphQLRequest): Request;
}

interface ClientArgsWithUrl extends ClientArgsCommon {
  url: string;
}

export type ClientArgs = ClientArgsWithUrl | ClientArgsWithBuildRequest;
```

(we need either `url` _or_ `buildRequest`)

Reply:
huh, yea, good point. I hadn't thought of that but it makes sense. I'll update

Reply:
On further consideration, I'm not totally sure this yields a ton of benefit, but does add some complexity to the actual client constructor. We can always revisit later, but for now I'm going to leave it as is.